src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -106,15 +106,35 @@ public DatabaseSearchService(IContext context, IFrameworkCompatibilityService fr
             };
         }
 
-        public Task<AutocompleteResponse> ListPackageVersionsAssync(
-            string packageId,
-            bool includePrerelease,
-            bool includeSemVer2,
+        public async Task<AutocompleteResponse> ListPackageVersionsAsync(
+            VersionsRequest request,
             CancellationToken cancellationToken)
         {
-            // TODO: Support versions autocomplete.
-            // See: https://github.com/loic-sharma/BaGet/issues/291
-            throw new NotImplementedException();
+            IQueryable<Package> search = _context.Packages;
+
+            if (!string.IsNullOrEmpty(request.Query))
+            {
+                var query = request.Query.ToLower();
+                search = search.Where(p => p.Id.ToLower().Equals(query));
+            }
```

Comment:
We can simplify this a little since the `id` query parameter is required for this API:

```suggestion
            var packageId = request.Query.ToLower();
            IQueryable<Package> search = _context
                .Packages
                .Where(p => p.Id.ToLower().Equals(query));
```